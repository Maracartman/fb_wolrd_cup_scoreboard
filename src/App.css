@font-face {
  font-family: "Ozone";
  src: local("Ozone"),
  url("./assets/fonts/Ozone/Ozone-xRRO.ttf") format("truetype");
  font-weight: normal;
}
