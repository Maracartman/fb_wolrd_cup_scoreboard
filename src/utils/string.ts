
export const transformToStringScore = (score: number, size?:number): string =>{
    const places = size ?? 2
    return String(score).padStart(places, '0')
}

const StringUtil = {
    transformToStringScore
}

export default StringUtil