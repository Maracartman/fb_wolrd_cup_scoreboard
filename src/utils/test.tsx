import React, { ReactElement } from 'react'
import {RenderOptions, render} from '@testing-library/react'
import TranslationsProvider from 'src/contexts/translations'

const AllTheProviders = ({children}:{children: ReactElement}): ReactElement => (
    <TranslationsProvider>
        {children}
    </TranslationsProvider>
)

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, {wrapper: AllTheProviders, ...options})

// re-export everything
export * from '@testing-library/react'

// override render method
export {customRender as render}