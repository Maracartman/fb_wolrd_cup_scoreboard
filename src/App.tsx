import React from 'react';
import ScoreboardPage from './components/Pages/Scoreboard/Scoreboard';
import './assets/fonts/Ozone/Ozone-xRRO.ttf';
import '@fontsource/roboto/300.css';
import '@fontsource/roboto/400.css';
import '@fontsource/roboto/500.css';
import '@fontsource/roboto/700.css';
import './App.css';
import TranslationsProvider from './contexts/translations';

/**
 * Here pages can be handled by navigators.
 */
const App = () => {
  return (
    <div className="App">
      <TranslationsProvider>
        <ScoreboardPage />
      </TranslationsProvider>
    </div>
  );
}

export default App;
