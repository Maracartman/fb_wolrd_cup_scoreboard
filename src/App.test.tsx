import { render, screen } from 'src/utils/test';
import App from './App';

test('renders learn react link', () => {
  const { container } = render(<App />);

  expect(container).toMatchSnapshot();
});
