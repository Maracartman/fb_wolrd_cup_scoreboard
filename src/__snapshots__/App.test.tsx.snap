// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div>
  <div
    class="App"
  >
    <div
      class="MuiGrid-root MuiGrid-container scoreboard-page css-11lq3yg-MuiGrid-root"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-grid-xs-12 scoreboard-container css-1nuw9uu-MuiGrid-root"
          >
            <div
              class="scoreboard-title MuiBox-root css-0"
            >
              Scoreboard
            </div>
            <div
              class="team-scores-container MuiBox-root css-0"
            >
              <div
                class="MuiContainer-root MuiContainer-maxWidthLg team-score-container css-1oqqzyl-MuiContainer-root"
              >
                <div
                  class="ts-data-box ts-status MuiBox-root css-0"
                >
                  Home
                </div>
                <div
                  class="ts-data-box country-name MuiBox-root css-0"
                >
                  ---
                </div>
                <div
                  class="ts-data-box MuiBox-root css-0"
                >
                  <div
                    class="score MuiBox-root css-0"
                  >
                    <div
                      class="score-text"
                    >
                      00
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiContainer-root MuiContainer-maxWidthLg team-score-container css-1oqqzyl-MuiContainer-root"
              >
                <div
                  class="ts-data-box ts-status MuiBox-root css-0"
                >
                  Away
                </div>
                <div
                  class="ts-data-box country-name MuiBox-root css-0"
                >
                  ---
                </div>
                <div
                  class="ts-data-box MuiBox-root css-0"
                >
                  <div
                    class="score MuiBox-root css-0"
                  >
                    <div
                      class="score-text"
                    >
                      00
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-grid-xs-12 css-1nuw9uu-MuiGrid-root"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthLg control-panel css-1oqqzyl-MuiContainer-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium css-ke5b6m-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Simulate a random Match
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium Mui-disabled MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
            disabled=""
            tabindex="-1"
            type="button"
          >
            Finish and add to summary
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
